calyx-frontend/src/attribute.rs

```diff
@@ -199,7 +199,7 @@ impl FromStr for Attribute {
 /// Inline storage for boolean attributes.
 pub(super) struct InlineAttributes {
     /// Boolean attributes stored in a 16-bit number.
-    attrs: u16,
+    pub attrs: u16,
```

Comment:
Rust tip: instead of making the underlying `attrs` public, let's `#[derive(PartialEq, Eq)]` on the `InlineAttributes` struct. Then you'll be able to do `self.inl == other.inl` without exposing the underlying representation.